### PR TITLE
feat: add obkv-hbase connector

### PR DESCRIPTION
**Pull request by whhe**
### Task Description

Add a new connector to sink data through OBKV HBase API.
 
### Solution Description

Ref: https://github.com/apache/flink-connector-hbase

### Passed Regressions

Locally tested.

### CODE REVIEW

**Comment by whhe** [comment]
@IHEII Please take a look on this PR, especially for [OBKVHBaseConnectionProvider](https://github.com/whhe/flink-connector-oceanbase/blob/obkv-hbase/flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/connection/OBKVHBaseConnectionProvider.java#L32-L38) related code.

File: flink-connector-obkv-hbase/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# flink-connector-oceanbase is licensed under Mulan PSL v2.
+# You can use this software according to the terms and conditions of the Mulan PSL v2.
+# You may obtain a copy of Mulan PSL v2 at:
+#         http://license.coscl.org.cn/MulanPSL2
+# THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
+# EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
+# MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+# See the Mulan PSL v2 for more details.
+
+com.oceanbase.connector.flink.OBKVHBaseDynamicTableSinkFactory
```
Comment (IHEII) [inline flink-connector-obkv-hbase/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory:11]: What is the purpose of this file?

File: flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/sink/OBKVHBaseStatementExecutor.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    private Put getPut(RowData rowData) {
+        Object rowKey = tableSchema.getRowKeyFieldGetter().getFieldOrNull(rowData);
+        if (rowKey == null) {
+            return null;
+        }
+        Put put = new Put((byte[]) (rowKey));
+        for (Map.Entry<String, Map<String, RowData.FieldGetter>> entry :
+                tableSchema.getFamilyQualifierFieldGetterMap().entrySet()) {
+            String familyName = entry.getKey();
+            byte[] family = Bytes.toBytes(familyName);
```
Comment (IHEII) [inline flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/sink/OBKVHBaseStatementExecutor.java:114]: How connector ensure that there is only one family?
(Users should be notified that we only support one family now if it is the user who guarantees this limitation.)

File: flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/connection/OBKVHBaseConnectionProvider.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    private transient OHTableClient tableClient;
+
+    public OBKVHBaseConnectionProvider(OBKVHBaseConnectionOptions options) {
+        this.options = options;
+    }
+
+    public HTableInterface getTable() throws Exception {
+        if (tableClient == null) {
+            tableClient = new OHTableClient(options.getTableName(), options.getConfig());
```
[the remark this answers is not in the document]
Comment (whhe) [reply flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/connection/OBKVHBaseConnectionProvider.java:34]: I plan to support it in another pr.

File: flink-connector-obkv-hbase/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory (hunk 1 above)
Comment (whhe) [reply flink-connector-obkv-hbase/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory:11]: This file is used to register it as one of DynamicTableSinkFactory implementation in Flink.

File: flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/sink/OBKVHBaseStatementExecutor.java (hunk 2 above)
Comment (whhe) [reply flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/sink/OBKVHBaseStatementExecutor.java:114]: There can be multiple families in one table, here multiple families will represent multiple entries in the familyQualifierFieldGetterMap.

File: flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/converter/OBKVHBaseRowConverter.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    private static final int MIN_TIME_PRECISION = 0;
+    private static final int MAX_TIME_PRECISION = 3;
+
+    @Override
+    protected FieldConverter createExternalConverter(LogicalType type) {
+        switch (type.getTypeRoot()) {
+            case BOOLEAN:
+                return (row, fieldIndex) -> Bytes.toBytes(row.getBoolean(fieldIndex));
+            case TINYINT:
+                return (row, fieldIndex) -> new byte[] {row.getByte(fieldIndex)};
```
[the remark this answers is not in the document]
Comment (whhe) [reply flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/converter/OBKVHBaseRowConverter.java:40]: Will have a try.

File: flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/sink/OBKVHBaseStatementExecutor.java [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+                    if (!deleteList.isEmpty()) {
+                        connectionProvider.getTable().delete(deleteList);
+                    }
+                    metricGroup.getIOMetricGroup().getNumRecordsOutCounter().inc(count);
+                    count = 0;
+                }
+            }
+            if (count > 0) {
+                connectionProvider.getTable().put(putList);
+                connectionProvider.getTable().delete(deleteList);
```
Comment (IHEII) [inline flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/sink/OBKVHBaseStatementExecutor.java:98]: It is more efficient for us to make all operations with the same family and qualifier set into the same putList and deleteList. Put operation using a buffer and sending requests depends on a unique strategy. Delete will directly send requests after you call delete().

> About Put

_HBASE_HTABLE_CLIENT_WRITE_BUFFER_ and _autoFlush_ is important.
- _HBASE_HTABLE_CLIENT_WRITE_BUFFER_: Max buffer size for put operation. Once the heap size reaches the max buffer size, the client will commit(send) all the put operations.
- _autoFlush_: Whether to commit(send) all the put operations after handling a putList(default true).

> About Delete

- Client will do multiple delete operations if you using delete(List<delete>), I guess we could try to make those delete into one delete to reduce the RPC.

File: flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/converter/OBKVHBaseRowConverter.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (whhe) [reply flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/converter/OBKVHBaseRowConverter.java:40]: Looks like `Bytes.toBytes` also use `new byte[]` to construct the response, to work with hbase connector [decoder](https://github.com/apache/flink-connector-hbase/blob/3ecbef9194b853781638b649f8ce1457a79e6b60/flink-connector-hbase-base/src/main/java/org/apache/flink/connector/hbase/util/HBaseSerde.java#L470), I think we should keep it here.

File: flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/connection/OBKVHBaseConnectionOptions.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        this.sysUsername = sysUsername;
+        this.sysPassword = sysPassword;
+    }
+
+    public String getTableName() {
+        return tableName;
+    }
+
+    public Configuration getConfig() {
+        org.apache.hadoop.conf.Configuration conf = new org.apache.hadoop.conf.Configuration();
```
[the remark this answers is not in the document]
Comment (whhe) [reply flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/connection/OBKVHBaseConnectionOptions.java:51]: You are right, I will add a setter for other properties.

**Pull request merged**